src/data/equiv/basic.lean

```diff
@@ -1688,11 +1688,22 @@ def range_of_left_inverse {α β : Sort*}
 
 Note that if `α` is empty, no such `f_inv` exists and so this definition can't be used, unlike
 the stronger but less convenient `equiv.set.range_of_left_inverse`. -/
+@[simps apply]
 abbreviation range_of_left_inverse' {α β : Sort*}
   (f : α → β) (f_inv : β → α) (hf : left_inverse f_inv f) :
   α ≃ set.range f :=
 range_of_left_inverse f (λ _, f_inv) (λ _, hf)
 
+@[simp] lemma range_of_left_inverse_symm_apply {α β : Sort*}
+  (f : α → β) (f_inv hf) (x : set.range f) :
+  (equiv.set.range_of_left_inverse f f_inv hf).symm x = f_inv ⟨classical.some x.2⟩ x :=
```

Comment:
Any reason not to let `simp` generate this one, which avoids using `classical`?

Reply:
Oh, I guess I should have implemented `range_of_left_inverse` using `nonempty_of_exists`:
```suggestion
  (equiv.set.range_of_left_inverse f f_inv hf).symm x = f_inv (nonempty_of_exists x.2) x :=
```

Reply:
`simps` gives a RHS containing `range_of_left_inverse._proof_2`, which doesn't seem so nice to me. Or should I put an explicit `exists.elim` in the definition of `range_of_left_inverse`?